### PR TITLE
Some modifications made to Hadge that were esential for running large scale processing of Onek1k data

### PR DESCRIPTION
**Pull request by maxozo**
This pull request addresses couple of things:
1) I have added a snippet from nf-core so that i can provide a profile for different institutes. Can run it as in: -profile sanger,cluster,singularity,test
2) There was a lot of redundant modules created by multi and single mode (please let me know if you do not agree with this strategy - but i double checked both single and multi and the modules are the same with only a slight modification). I think its better for a long term maintenance of the pipeline to rely on just one branch, so so same vireo module is used both in single and multi mode. this cleans up the pipeline significantly.
3) I have changed to outputdirectory paths. Output paths shouldnt contain $projectDir - this causes issues providing full paths and places results wherever hadge is cloned. 
4) freebayes takes ages to run. for this reason i added an optional subsetting module that preprocesses bam and removes the variants that we would not use in deconvolutions anyways. for OneK1k i subset the bams down to cellsnp comon variants to speed up the pipeline.
5) there was a conflicts with the output directories in genetics summary module. the logic of input chanels in this module had to be changed.

### CODE REVIEW

File: bin/bff.R [hunk 1]
```diff
@@ -4,11 +4,21 @@ library(DropletUtils)
 library(Seurat)
 library(ggplot2)
 library(cowplot)
+if(!require("cellhashR")){
```
Comment (Zethson) [inline bin/bff.R:7]: CC @mari-ga

File: modules/multi/gene_demultiplexing.nf [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+    input:
+        tuple val(sampleId), path(sam), path(sam_index), path(barcodes)
+        path vcf
+
+    output:
+        tuple val(sampleId), path("${sampleId}__filtered_bam_file.bam"), path("${sampleId}__filtered_bam_file.bam.csi"), emit: input
+    
+    script:
+        """ 
+
```
Comment (Zethson) [inline modules/multi/gene_demultiplexing.nf:38]:
```suggestion
```

File: modules/multi/hash_demulti/demuxem.nf [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 
-process demuxem {
-    publishDir "$projectDir/$params.outdir/$sampleId/$params.mode/hash_demulti/demuxem", mode:'copy'
+process demuxem{
+    publishDir "$params.outdir/$sampleId/$params.mode/hash_demulti/demuxem", mode:'copy'
     label 'small_mem'
 
-    conda 'bioconda::pegasuspy bioconda::scanpy bioconda::demuxEM'
-
+    conda "bioconda::pegasuspy demuxEM scanpy"
```
Comment (Zethson) [inline modules/multi/hash_demulti/demuxem.nf:8]: scanpy should be taken from conda-forge

File: subworkflows/HADGE.nf [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    scSplit_out_ch = scSplit_out.flatten().map{r1-> tuple(    "$r1".replaceAll(".*scsplit_",""), r1 )}
+    souporcell_out_ch = souporcell_out.flatten().map{r1-> tuple(    "$r1".replaceAll(".*souporcell_",""), r1 )}
+
+    summary_input = input_list_summary.join(souporcell_out_ch,by:0,remainder: true).join(scSplit_out_ch,by:0,remainder: true).join(vireo_out_ch,by:0,remainder: true).join(freemuxlet_out_ch,by:0,remainder: true).join(demuxlet_out_ch,by:0,remainder: true)
+    summary_input = summary_input.filter{ it[0] != 'no_result' }
+
+    summary(summary_input,
+            params.generate_anndata, params.generate_mudata)
+
+    // Channel.fromPath(params.multi_input) \
```
Comment (Zethson) [inline subworkflows/HADGE.nf:70]: Can we get rid of this section?

**Review by Zethson — COMMENTED** [review]
Thank you!

File: modules/multi/hash_demulti/demuxem.nf (hunk 3 above)
Comment (maxozo) [reply modules/multi/hash_demulti/demuxem.nf:8]: ahh, good point. maybe this causes my issues.

**Review by wxicu — APPROVED** [review]
Thank you!!! I adaped the code to the hashing workflow. Everything looks good now.

**Pull request merged**